app/src/config/smmtConfigurationLoader.js

```diff
@@ -0,0 +1,45 @@
+const co = require('co');
+const AWS = require('../wrapper/awsSdkWrapper');
+const loggerFactory = require('../logger/createLogger');
+const env = require('../wrapper/envVariablesWrapper');
+
+const logger = loggerFactory.create();
+const kms = new AWS.KMS();
+
+let smmtApiKey;
+
+function getSmmtApiKey() {
+  return kms.decrypt({
+    CiphertextBlob: Buffer.from(env.SMMT_API_KEY, 'base64'),
+  }).promise()
+    .then((data) => {
+      logger.info('SMMT api key decrypted properly.');
+
+      return data.Plaintext.toString('ascii');
+    })
+    .catch((reason) => {
+      logger.error({ context: reason }, 'SMMT api key decryption failed.');
+      return undefined;
+    });
+}
+
+exports.load = () => co.wrap(function* loadConfig(key) {
```

Comment:
loadConfig(key) does not seem to be clear enough to me... what's the purpose of key argument ?

Reply:
key parameter is used by unit tests. it allows to verify if KMS will not be called if key was already fetched.